lib/sender.js

```diff
@@ -67,67 +68,91 @@ module.exports = {
 */
 function send(client) {
     return function send(message, options) {
-        return new Promise((resolve, reject) => {
-            options = options || {};
+        const body = composeSenderRequestBoddyForOneMessage(message, options);
+        return makeRequest(client, body);
+    };
+}
+
+/**
+    The sendAll Method for multiple messages
+    @param {Array}  messages - An array of objects containing the fields described in send Method [{message, options}, ... ]
+*/
+function sendAll(client) {
+    return function sendAll(messages) {
+        const body = messages.map(x => composeSenderRequestBoddyForOneMessage(x.message, x.options));
+        client.url += "batch";
+        return makeRequest(client, body);
+    };
+}
 
-            const newMessage = {};
-            // Make sure we remove empty arrays for criteria
-            newMessage.criteria = helpers.removeEmptyArrays(options.criteria);
-            newMessage.config = options.config;
+function composeSenderRequestBoddyForOneMessage(message, options) {
+    options = options || {};
 
-            for(let option in options) {
-                if(option !== 'criteria' || option !== 'config') {
-                    newMessage[option] = options[option];
-                }
-            }
+    const newMessage = {};
 
-            for(let key in message ) {
-                if (key === 'apns') {
-                    for(let apnsKey in message[key]) {
-                        if(apnsKey === 'actionCategory' || apnsKey === 'contentAvailable' || apnsKey === 'urlArgs' || apnsKey === 'titleLocKey' || apnsKey === 'titleLocArgs') {
-                            message[key][helpers.camelToDash(apnsKey)] = message[key][apnsKey];
-                            delete message[key][apnsKey];
-                        }
-                    }
-                }
+    // Make sure we remove empty arrays for criteria
+    newMessage.criteria = helpers.removeEmptyArrays(options.criteria);
+    newMessage.config = options.config;
+
+    for (let option in options) {
+        if (option !== 'criteria' || option !== 'config') {
+            newMessage[option] = options[option];
+        }
+    }
 
-                if(key === 'simplePush' || key === 'userData') {
-                    message[helpers.camelToDash(key)] = message[key];
-                    delete message[key];
+    for (let key in message) {
+        if (key === 'apns') {
+            for (let apnsKey in message[key]) {
+                if (apnsKey === 'actionCategory' || apnsKey === 'contentAvailable' || apnsKey === 'urlArgs' || apnsKey === 'titleLocKey' || apnsKey === 'titleLocArgs') {
+                    message[key][helpers.camelToDash(apnsKey)] = message[key][apnsKey];
+                    delete message[key][apnsKey];
                 }
             }
+        }
 
-            newMessage.message = message;
+        if (key === 'simplePush' || key === 'userData') {
```

Comment:
Oh, :-) 
The `simplePush` option is not supported anymore, let's not add some code in the new function for that